### ISSUE TITLE
howto customize dovecot config

### ISSUE THREAD

**Issue by thy1985** [hardware/mailserver#186]
hi there

first off - thanks to the maintainer and all contributors of this repository! this is really a time saver!

I am migrating over from a gentoo-based similar setup (just SpamAssasin instead of rspamd, courier instead of dovecot) to ease a future box swap.

It is working just fine. Only thing I had to tweak in order to make the transistion transparent to mail user was to change the namespace settings of file /etc/dovecot/conf.d/10-mail.conf from (to match courier defaults)
```
namespace inbox {
  inbox = yes
}
```

to

```
namespace inbox {
  prefix = INBOX.
  separator = .
  inbox = yes
}
```

Now, as I am pretty new to both docker and dovecot, I am asking myself how I can make that persistent over container recreation without breaking the ability to upgrade later.

I considered so far:
1. Create my own image FROM the original and COPY the changed file during build time (or add an overwrite as 99-custom.conf, would that work?)
2. "mount" a single file in the volume section to overwrite (or add with the 99-idea) the file in question
 
But both add an extra step to upgrades: Either build a new own image or check if anything else changed in the file in the new image and adapt my own copy of the file.

So basically my question is: Is there a better way "to inject" such a config tweak into a container?

Any pointer would be highly appreciated! Thanks a lot in advance!

Cheers,

**Comment by hardware** [comment]
For this kind of configuration changes, the best way is to override the `10-mail.conf` configuration file.

```yml
volumes:
  - /path/to/10-mail.conf:/etc/dovecot/conf.d/10-mail.conf
```

```ini
mail_plugins = $mail_plugins quota
mail_location = maildir:/var/mail/vhosts/%d/%n/{{ VMAIL_SUBDIR }}
maildir_stat_dirs=yes

namespace inbox {
  prefix = INBOX.
  separator = .
  inbox = yes
}

mail_uid = {{ VMAILUID }}
mail_gid = {{ VMAILGID }}
mail_privileged_group = vmail
```

To extend dovecot configuration, I can add `!include_try custom.conf` in `dovecot.conf` but this will not work in your case because you want to make changes and not extend the configuration.

> or check if anything else changed in the file in the new image and adapt my own copy of the file.

`10-mail.conf` should not be changed noticeably without a new version of this docker image. So it's pretty safe to override it.

**Comment by thy1985** [comment]
Thanks @hardware for your reply! So I'll document to compare those files when there is  new docker image version.
I had to overwrite two more files to make it work with my changes (added INBOX. namespace before Trash and Spam):
```
      - /mnt/docker/mail/dovecot-custom/10-mail.conf:/etc/dovecot/conf.d/10-mail.conf
      - /mnt/docker/mail/dovecot-custom/90-quota.conf:/etc/dovecot/conf.d/90-quota.conf
      - /mnt/docker/mail/dovecot-custom/90-sieve.conf:/etc/dovecot/conf.d/90-sieve.conf
```

**Comment by thy1985** [comment]
well, one more thing. I noticed I need to tweak /var/mail/sieve/default.sieve, too.
Spam is INBOX.Spam so it currently fails to save to Spam:
```
 failed to store into mailbox 'Spam': Mailbox doesn't exist: Spam
```

But that one seems to be recreated upon image start. How do I persist a change there?

**Comment by hardware** [comment]
https://github.com/hardware/mailserver/issues/177

**Comment by thy1985** [comment]
Thanks. In analogy to the configuration tweaks above I considered #177 to be ADDING to config. But if that means custom.sieve overwrites default.sieve that is what I want

**Comment by hardware** [comment]
custom.sieve overwrites default.sieve :)

**Comment by thy1985** [comment]
Perfect, thanks again
